mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -258,6 +261,54 @@ def bind(self, bus):
             # Initialize the SkillGui
             self.gui.setup_default_handlers()
 
+            self._register_public_api()
+
+    def _register_public_api(self):
+        """ Find and register api methods.
+        Api methods has been tagged with the api_method member, for each
+        method where this is found the method a message bus handler is
+        registered.
+        Finally create a handler for fetching the api info from any requesting
+        skill.
+        """
+
+        def wrap_method(func):
+            """Boiler plate for returning the response to the sender."""
+            def wrapper(message):
+                result = func(*message.data['args'], **message.data['kwargs'])
+                self.bus.emit(message.response(data={'result': result}))
+
+            return wrapper
+
+        methods = [a for a in get_non_properties(self)
+                   if hasattr(a, '__name__')]
```

Comment:
None of my decorated methods seem to have a `__name__` attribute. However they do get assigned the `api_method`. 

Example Skill with an API method: https://github.com/krisgesling/skill-api-source-skill

Reply:
I get `True` If I do `self.log.info(hasattr(self.speak_from_other_skill, '__name__'))` on your skill

Reply:
The check does look a bit wonky though. Let's see if I can improve it...

Reply:
Yeah, same if I log from the Skill itself, but logging from here returns False:
```
if self.skill_id == 'skill-api-source-skill':
            for a in get_non_properties(self):
                LOG.info("{} has __name__: {}".format(a, hasattr(a, '__name__')))
```

Reply:
Ahh, the get_non_properties() returns a string not a reference to the attribute, so looks like we need this to be:
```
        methods = [a for a in get_non_properties(self)
                   if hasattr(getattr(self, a), '__name__')]
```

Reply:
Ah great catch...how has this ever worked?

Reply:
Pushed your suggested fix and updated the CLI code to handle :api without a skill argument